File: app/src/main/java/life/plank/juna/zone/view/adapter/post/CommentReplyAdapter.java
package life.plank.juna.zone.view.adapter.post;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.RequestManager;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import life.plank.juna.zone.R;
import life.plank.juna.zone.data.model.FeedItemCommentReply;

import static life.plank.juna.zone.util.UIDisplayUtil.getDp;

public class CommentReplyAdapter extends RecyclerView.Adapter<CommentReplyAdapter.PostCommentReplyViewHolder> {

    private RequestManager glide;
    private List<FeedItemCommentReply> replies;

    public CommentReplyAdapter(RequestManager glide, List<FeedItemCommentReply> replies) {
        this.glide = glide;
        this.replies = replies;
    }

    @NonNull
    @Override
    public PostCommentReplyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_base_comment, parent, false);
        return new PostCommentReplyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PostCommentReplyViewHolder holder, int position) {
        holder.commentTextView.setText(replies.get(position).getMessage());
        holder.profileNameTextView.setText(replies.get(position).getCommenterDisplayName());
        glide.load(replies.get(position).getCommenterProfilePicUrl())
                .apply(RequestOptions.overrideOf((int) getDp(20), (int) getDp(20)))
                .into(holder.profilePic);
        
    }

    @Override
    public int getItemCount() {
        return replies.size();
    }

    static class PostCommentReplyViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.profile_pic)
        ImageView profilePic;
        @BindView(R.id.profile_name_text_view)
        TextView profileNameTextView;
        @BindView(R.id.comment_text_view)
        TextView commentTextView;

        PostCommentReplyViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/adapter/post/CommentReplyAdapter.kt
package life.plank.juna.zone.view.adapter.post

import android.text.SpannableStringBuilder
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.RequestManager
import com.bumptech.glide.request.RequestOptions
import kotlinx.android.synthetic.main.item_base_comment.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.data.model.FeedItemComment
import life.plank.juna.zone.util.common.AppConstants
import life.plank.juna.zone.util.common.DataUtil.isNullOrEmpty
import life.plank.juna.zone.util.common.color
import life.plank.juna.zone.util.common.formatMentions
import life.plank.juna.zone.util.common.semiBold
import life.plank.juna.zone.util.facilis.onDebouncingClick
import life.plank.juna.zone.util.time.DateUtil
import life.plank.juna.zone.util.view.UIDisplayUtil.getDp
import life.plank.juna.zone.view.fragment.base.BaseCommentContainerFragment
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread

class CommentReplyAdapter(
        private val glide: RequestManager,
        private val parentComment: FeedItemComment,
        private val parentCommentPosition: Int,
        private val replies: List<FeedItemComment>,
        private val commentContainerFragment: BaseCommentContainerFragment
) : RecyclerView.Adapter<CommentReplyAdapter.PostCommentReplyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostCommentReplyViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_base_comment, parent, false)
        return PostCommentReplyViewHolder(view)
    }

    override fun onBindViewHolder(holder: PostCommentReplyViewHolder, position: Int) {
        holder.itemView.run {
            val reply = replies[position]
            doAsync {
                //            TODO: remove the if block below when backend completes forum comment mentions
                if (isNullOrEmpty(reply.commenterHandle)) {
                    reply.commenterHandle = reply.commenterDisplayName.replace(" ", "")
                }
                val formattedReply =
                        SpannableStringBuilder(reply.commenterHandle.semiBold().color(R.color.black))
                                .append(AppConstants.SPACE)
                                .append(reply.message.formatMentions())
                uiThread { comment_text_view.text = formattedReply }
            }

            comment_time_text.text = DateUtil.getCommentDateAndTimeFormat(reply.time)

            glide.load(reply.commenterProfilePictureUrl)
                    .apply(RequestOptions.overrideOf(getDp(20f).toInt(), getDp(20f).toInt()))
                    .into(profile_pic!!)

            reply_text_view.onDebouncingClick {
                commentContainerFragment.replyAction(reply_text_view, holder.itemView, reply.commenterHandle, parentComment, parentCommentPosition, position)
            }

            like_text_view.onDebouncingClick {}
        }
    }

    override fun getItemCount(): Int = replies.size

    class PostCommentReplyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}